src/core/wallets/btc/bitget/provider.ts

```diff
@@ -0,0 +1,206 @@
+import { Psbt } from "bitcoinjs-lib";
+
+import type { BTCConfig, Fees, InscriptionIdentifier, UTXO, WalletInfo } from "@/core/types";
+import { Network } from "@/core/types";
+import { validateAddress } from "@/core/utils/wallet";
+import { BTCProvider } from "@/core/wallets/btc/BTCProvider";
+
+const INTERNAL_NETWORK_NAMES = {
+  [Network.MAINNET]: "livenet",
+  [Network.TESTNET]: "testnet",
+  [Network.SIGNET]: "signet",
+};
+
+export class BitgetProvider extends BTCProvider {
+  private provider: any;
+  private walletInfo: WalletInfo | undefined;
+
+  constructor(wallet: any, config: BTCConfig) {
+    super(config);
+
+    // check whether there is an Bitget Wallet extension
+    if (!wallet?.unisat) {
```

Comment:
> hmmm.... so bitget is using unisat namespace for btc? Could we ping them to fix this? otherwise we gonna have problems when onboard unisat
No, it's `window.bitkeep.unisat` not `window.unisat`